### ISSUE TITLE
about [Gaussian_yolo]

### ISSUE THREAD

**Issue by Code-Fight** [AlexeyAB/darknet#4190]
Hi, @AlexeyAB 
Thanks for you share.
I found you add "[Gaussian_yolo]"
Can I train now？Any precautions？

Trhanks

**Comment by AlexeyAB** [comment]
@Code-Fight Hi, yes.

just use `[Gaussian_yolo]` instead of `[yolo]` in cfg-file

**Comment by Code-Fight** [comment]
Hi @AlexeyAB ，Thanks for your reply.
I got it
However, when I went to train myself , the network could not converge and fluctuated around 10 rounds.
Excuse me, what is the reason?
Single card, 1080ti, lr=0.001
Thanks

**Comment by zpmmehrdad** [comment]
@AlexeyAB Hi,

Is Gaussian_yolo layer support for the all cfg's?
SPP, TinyPRN and etc.

**Comment by Code-Fight** [comment]
Hi @AlexeyAB ,Thanks
This  is my cfg and chart.png

![chart](https://user-images.githubusercontent.com/8985847/67920606-c9986f80-fbdf-11e9-8d10-aae4e6587165.png)

[sword_full_size_Gaussian.cfg.txt](https://github.com/AlexeyAB/darknet/files/3792237/sword_full_size_Gaussian.cfg.txt)

**Comment by AlexeyAB** [comment]
Try to set `learning_rate=0.0001` instead of `learning_rate=0.001` and train

**Comment by Code-Fight** [comment]
Hi @AlexeyAB   I got it , I will train again
Thank you very much

**Comment by AlexeyAB** [comment]
@zpmmehrdad 

For `[Gaussian_yolo]` should be:
`filters  = (classes + 8 + 1) * <numbers in mask>`

instead of (for [yolo]):
`filters  = (classes + 4 + 1) * <numbers in mask>]`

**Comment by Code-Fight** [comment]
HI ,@AlexeyAB 
 I train again.
learning_rate=0.0001 instead of learning_rate=0.001
but, still the same problem.



![0191031182523](https://user-images.githubusercontent.com/8985847/67939436-81437680-fc0c-11e9-93a7-9fa50ce73ec6.jpg)

**Comment by AlexeyAB** [comment]
This is normal for [Gaussian_yolo] layer. Check the mAP.
Or try to use even `learning_rate=0.00001`

**Comment by Code-Fight** [comment]
@AlexeyAB  Thanks , I will try again

**Comment by Code-Fight** [comment]
Hi @AlexeyAB  ,I train again with "map" .
I found the loss increase and "map" increase too.
Do you know what the reason is?
Thank you very much
![chart](https://user-images.githubusercontent.com/8985847/67995036-57796680-fc83-11e9-8848-b4705d09acf4.png)

**Comment by AlexeyAB** [comment]
@Code-Fight I don't know, may be delta of uncertainty is increasing https://github.com/AlexeyAB/darknet/blob/master/src/gaussian_yolo_layer.c#L182-L185

**Comment by Code-Fight** [comment]
@AlexeyAB  I got it
Thanks for your help
Thanks

**Comment by zpmmehrdad** [comment]
@Code-Fight  Hi,

What cfg did you use?

**Comment by Code-Fight** [comment]
Hi @zpmmehrdad 
this cfg [yolov3.cfg](https://github.com/AlexeyAB/darknet/blob/4d9addedd719cfb5d68cb3b39d5c7f9193071c0d/cfg/yolov3.cfg#L1)
and changed [Yolo] to [Gussian_Yolo]

**Comment by zpmmehrdad** [comment]
@Code-Fight Thanks, Did you check [Gussian_Yolo] with SPP?

**Comment by Code-Fight** [comment]
@zpmmehrdad no, i didn't

**Comment by zpmmehrdad** [comment]
@Code-Fight Hi,

Could you share the mAP and FPS with [Gussian_Yolo] and with [Yolo] on your dataset?

**Comment by litingsjj** [comment]
@AlexeyAB Have you tried gaussian with yolov2?

**Comment by lq0104** [comment]
> Hi @AlexeyAB ,I train again with "map" .
> I found the loss increase and "map" increase too.
> Do you know what the reason is?
> Thank you very much
> ![chart](https://user-images.githubusercontent.com/8985847/67995036-57796680-fc83-11e9-8848-b4705d09acf4.png)

hi @Code-Fight would you show me your final learning rate, thx~

**Comment by Code-Fight** [comment]
@lq0104 Hi，this is my cfg
-------------------------------------------------------
batch=64
subdivisions=16
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1
flip=0
learning_rate=0.00001
burn_in=1000

**Comment by lq0104** [comment]
Yesterday I train my dataset with gaussian layer, as a result after hundreds of iterators the loss diverge to nan with learning rate=0.001, thanks again and i will try the new learning rate.  :）

**Comment by Code-Fight** [comment]
@zpmmehrdad HI,sorry,my reply too late.
[Gussian_Yolo]  FPS 55+   1080ti

**Comment by lsydd** [comment]
Should I changed the label when I using [Gaussian_yolo] instead of [yolo]?
